applications/CompressiblePotentialFlowApplication/custom_elements/incompressible_adjoint_potential_flow_element.cpp

```diff
@@ -0,0 +1,558 @@
+    {
+        mpPrimalElement->CalculateLeftHandSide(rLeftHandSideMatrix,
+                                               rCurrentProcessInfo);
+    }
+
+    template <int Dim, int NumNodes>
+    void IncompressibleAdjointPotentialFlowElement<Dim, NumNodes>::CalculateRightHandSide(VectorType& rRightHandSideVector,
+                                        ProcessInfo& rCurrentProcessInfo) 
+    {
+        mpPrimalElement->CalculateRightHandSide(rRightHandSideVector,
+                                                rCurrentProcessInfo);
+    }
+
+    template <int Dim, int NumNodes>
+    void IncompressibleAdjointPotentialFlowElement<Dim, NumNodes>::CalculateFirstDerivativesContributions(MatrixType& rLeftHandSideMatrix,
+							VectorType& rRightHandSideVector,
+							ProcessInfo& rCurrentProcessInfo) 
+    {
+        mpPrimalElement->CalculateFirstDerivativesContributions(rLeftHandSideMatrix,
+							rRightHandSideVector,
+							rCurrentProcessInfo);
+    }
+
+    template <int Dim, int NumNodes>
+    void IncompressibleAdjointPotentialFlowElement<Dim, NumNodes>::CalculateFirstDerivativesLHS(MatrixType& rLeftHandSideMatrix,
+					        ProcessInfo& rCurrentProcessInfo) 
+    {
+        mpPrimalElement->CalculateFirstDerivativesLHS(rLeftHandSideMatrix,
+					        rCurrentProcessInfo);
+    }
+
+    template <int Dim, int NumNodes>
+    void IncompressibleAdjointPotentialFlowElement<Dim, NumNodes>::CalculateFirstDerivativesRHS(VectorType& rRightHandSideVector,
+					      ProcessInfo& rCurrentProcessInfo) 
+    {
+        mpPrimalElement->CalculateFirstDerivativesRHS(rRightHandSideVector,
+					        rCurrentProcessInfo);
+    }
+
+    template <int Dim, int NumNodes>
+    void IncompressibleAdjointPotentialFlowElement<Dim, NumNodes>::CalculateSecondDerivativesContributions(MatrixType& rLeftHandSideMatrix,
+							 VectorType& rRightHandSideVector,
+							 ProcessInfo& rCurrentProcessInfo) 
+    {
+        mpPrimalElement->CalculateSecondDerivativesContributions(rLeftHandSideMatrix,
+							    rRightHandSideVector,
+							    rCurrentProcessInfo);
+    }
+
+    template <int Dim, int NumNodes>
+    void IncompressibleAdjointPotentialFlowElement<Dim, NumNodes>::CalculateSecondDerivativesLHS(MatrixType& rLeftHandSideMatrix,
+					       ProcessInfo& rCurrentProcessInfo) 
+    {
+        mpPrimalElement->CalculateSecondDerivativesLHS(rLeftHandSideMatrix,
+					        rCurrentProcessInfo);
+    }
+
+    template <int Dim, int NumNodes>
+    void IncompressibleAdjointPotentialFlowElement<Dim, NumNodes>::CalculateSecondDerivativesRHS(VectorType& rRightHandSideVector,
+					       ProcessInfo& rCurrentProcessInfo) 
+    {
+        mpPrimalElement->CalculateSecondDerivativesRHS(rRightHandSideVector,
+					        rCurrentProcessInfo);
+    }
+
+    template <int Dim, int NumNodes>
+    void IncompressibleAdjointPotentialFlowElement<Dim, NumNodes>::CalculateMassMatrix(MatrixType& rMassMatrix, ProcessInfo& rCurrentProcessInfo) 
```

Comment:
Do you really need the functions CalculateMassMatrix, CalculateDampingMatrix, CalculateSecondDerivativesLHS, CalculateFirstDerivativesRHS, CalculateFirstDerivativesLHS and CalculateFirstDerivativesContributions? Are you calling them somewhere?

Reply:
I am not using them! At first I wasn't sure what functions I would need so I included all of them. I updated the files with the ones that I currently need.